### PR TITLE
KAFKA-2698: add paused API

### LINKED ISSUE #593: KAFKA-2698: Add paused() method to KafkaConsumer

### PR DESCRIPTION
**Pull request by zackdever**
closes #593

### CODE REVIEW

File: kafka/consumer/group.py [hunk 1]
```diff
@@ -527,6 +527,14 @@ def pause(self, *partitions):
             log.debug("Pausing partition %s", partition)
             self._subscription.pause(partition)
 
+    def paused(self):
+        """Get the partitions that were previously paused by a call to pause().
+
+        Returns:
+            set: {partition (TopicPartition), ...}
+        """
+        return self._subscription.paused_partitions()
```
Comment (zackdever) [inline kafka/consumer/group.py]: The java client returns a `Collections.unmodifiableSet` here. We could return a `frozenset`, but I stuck with `set` to be consistent with `assignment()` and `subscription()`.

File: test/test_consumer_group.py [hunk 2]
```diff
@@ -120,6 +122,24 @@ def consumer_thread(i):
...
+@pytest.mark.skipif(not version(), reason="No KAFKA_VERSION set")
+def test_paused(kafka_broker, topic):
+    consumer = KafkaConsumer(bootstrap_servers=get_connect_str(kafka_broker))
+    topics = [TopicPartition(topic, 1)]
+    consumer.assign(topics)
+    assert set(topics) == consumer.assignment()
+    assert set() == consumer.paused()
+
+    consumer.pause(topics[0])
+    assert set(topics) == consumer.paused()
```
Comment (dpkp) [inline test/test_consumer_group.py]: `set([topics[0]])` ?

Comment (zackdever) [inline test/test_consumer_group.py]: that's the same thing, correct?

**Comment by dpkp** [comment]
Thanks!

**Pull request merged**